File: Review-11-Python3-Shapes/Example-1/shapes/RightTriangle.py
import math
import copy

from shapes.Shape import Shape
from shapes.Triangle import Triangle


class RightTriangle(Triangle):
    """
    A Triangle with a hypotenuse and height as two
    of its sides
    """

    ONE_HALF = 1.0 / 2.0  # @f$ \frac{1}{2} @f$

    @staticmethod
    def __computeHypotenuse(base, height):
        """
        Compute the hyptoenuse using:
        @f$ hypotenuse = \sqrt{base^2 + height^2} @f$
        :base: the base of a Right Triangle
        :param: height the height of a Right Triangle

        :return: hypotenuse of a right triangle
        """
        return math.sqrt(
            (base ** 2) + (height ** 2)
        )

    def __init__(self, base=1, height=1):
        """
        Construct a RightTriangle
        with base and height set to 1 by
        default.
        """
        self.name = "Right Triangle"

        self.side_a = base
        self.side_b = height
        self.side_c = RightTriangle.__computeHypotenuse(
            self.side_a,
            self.side_b
        )

    @property
    def base(self):
        return self.side_a

    @base.setter
    def base(self, side):

        self.side_a = side

        self.side_c = RightTriangle.__computeHypotenuse(
            self.side_a,
            self.side_b
        )

    @property
    def height(self):
        self.side_b

    @height.setter
    def height(self, side):
        self.side_b = side

        self.side_c = RightTriangle.__computeHypotenuse(
            self.side_a,
            self.side_b
        )

    @property
    def hypotenuse(self):
        return self.side_c

    def area(self):
        """
        Compute the area using
        @f$ Area = \frac{1}{2}*base*height @f$

        :return: the area
        """

        return RightTriangle.ONE_HALF * self.side_a * self.side_b

    def __deepcopy__(self, memo):
        """
        Return a new duplicate Shape
        """

        return RightTriangle(
            copy.deepcopy(self.side_a),
            copy.deepcopy(self.side_b),
        )

    def __str__(self):
        formatStr = (
            "{:<" +
            str(Shape.WIDTH_LABEL) +
            "}:{:>" +
            str(Shape.WIDTH_VALUE) +
            ".4f}\n"
        )

        return (
            super(Triangle, self).__str__() +
            formatStr.format("Base", self.side_a) +
            formatStr.format("Height", self.side_b) +
            formatStr.format("Hypotenuse", self.side_c) +
            formatStr.format("Perimeter", self.perimeter()) +
            formatStr.format("Area", self.area())
        )

if __name__ == "__main__":

    s = RightTriangle(1, 2)
    print(s)

File: Review-08-Java-Shapes/Example-8/src/main/java/edu/odu/cs/cs330/examples/shapes/RightTriangle.java
// Thomas Kennedy
// CS 330 Fall 2014

package edu.odu.cs.cs330.examples.shapes;

import java.util.Scanner;

/**
 * A Triangle with all sides set to the same length
 *
 * @author Thomas J Kennedy
 */
public class RightTriangle extends Triangle implements Cloneable {
    /**
     * A constant for one half
     */  
    private static final double ONE_HALF = 1.0/2.0; ///< @f$ \frac{1}{2} @f$
    
    /**
     * Compute the hyptoenuse using: 
     * @f$ hypotenuse = \sqrt{base^2 + height^2} @f$
     *
     * @param base the base of a Right Triangle
     * @param height the height of a Right Triangle
     *
     * @return hypotenuse of a right triangle
     */
    private static double computeHypotenuse( double base, double height )
    {
        return Math.sqrt(( base * base ) + ( height * height ));
    }

    /**
     * Construct a RightTriangle
     * with base and height set to 1.
     */
    public RightTriangle()
    {
        super();
        _name   = "Right Triangle";    

        _side_c = computeHypotenuse(
            _side_a,
            _side_b
        );
    }

    /**
     * Construct a RightTriangle
     *
     * @param base the desired base
     * @param height the desired height
     */
    public RightTriangle( double base, double height )
    {
        this._name   = "Right Triangle";

        this._side_a = base;
        this._side_b = height;
        this._side_c = RightTriangle.computeHypotenuse(
            _side_a,
            _side_b
        );
    }

    /**
     * Construct a RightTriangle
     *
     * @param src the RightTriangle to copy       
     */ 
    public RightTriangle( RightTriangle src )
    {
        this._name   = src._name;

        this._side_a = src._side_a;
        this._side_b = src._side_b;
        this._side_c = src._side_c;
    }

    /**
     * Return the base
     */
    public double base()
    {
        return _side_a;
    }

    /**
     * Modify the base
     *
     * @param side the replacement base
     */
    public void base( double side )
    {
        _side_a = side;
        
        _side_c = RightTriangle.computeHypotenuse(
            _side_a,
            _side_b
        );
    }

    /**
     * Return the height
     */
    public double height()
    {
        return _side_b;
    }

    /**
     * Modify the height
     *
     * @param side the replacement height
     */
    public void height( double side )
    {
        _side_b = side;

        _side_c = RightTriangle.computeHypotenuse(
            _side_a,
            _side_b
        );
    }

    /**
     * Return the hypotenuse
     *
     * @return the hypotenuse
     */
    public double hypotenuse()
    {
        return _side_c;
    }

    /**
     * Compute the area using
     * @f$ Area = \frac{1}{2}*base*height @f$
     *
     * @return the area
     */
    public double area()
    {
        return ONE_HALF * _side_a * _side_b;
    }

    /**
     * Return a new duplicate RightTriangle
     */
    @Override
    public Object clone() throws CloneNotSupportedException
    {
        return new RightTriangle( this );
    }

    /**
     * Read the shape
     *
     * @param scanner the input stream--scanner in this example
     */
    @Override
    public void read(Scanner scanner)
    {
        this.base(scanner.nextDouble());
        this.height(scanner.nextDouble());

        this._side_c = computeHypotenuse(this.base(), this.height());
    }

    /**
     * Print the RightTriangle
     */
    @Override
    public String toString()
    {
        return (
            String.format( getFormat( "s\n" ),   "Name",       this._name        ) +
            String.format( getFormat( ".4f\n" ), "Base",       this.base()       ) +
            String.format( getFormat( ".4f\n" ), "Height",     this.height()     ) +
            String.format( getFormat( ".4f\n" ), "Hypotenuse", this.hypotenuse() ) +
            String.format( getFormat( ".4f\n" ), "Perimeter",  this.perimeter()  ) +
            String.format( getFormat( ".4f\n" ), "Area",       this.area()       )
        );
    }
}





